app/lib/actions/katello/content_view_puppet_environment/create.rb

```diff
@@ -4,15 +4,13 @@ module ContentViewPuppetEnvironment
       class Create < Actions::EntryAction
         def plan(puppet_environment, clone = false)
           internal_capsule = SmartProxy.default_capsule
-          fail _("Content View %s  cannot be published without an internal capsule." % puppet_environment.name) unless internal_capsule
+          puppet_path = internal_capsule.present? ? puppet_environment.generate_puppet_path(internal_capsule) : nil
```

Comment:
If this is nil, what happens in the Repository::Create action below given no path would be specified?

Reply:
Without a path we already create it without the `puppet_install_distributor`. Which is what we want when the Pulp nodes isn't also a Puppet Master.
See https://github.com/Katello/katello/blob/master/app/lib/actions/pulp/repository/create.rb#L120

Reply:
For the server side case, could this lead to some confusion for users? Say for some reason, whether it be the server does not have an internal proxy that it knows about or a user did not install a puppet master/server, and they add puppet modules to their CV this won't generate the output from that anywhere. Now perhaps that is to be expected given no Puppet server/master is present, but I could see it confusing people without any output or warning.

Still wrapping my head around this but I am wondering:

 1) What happens if you  move the default proxy off to another box co-located with Puppet master?

Reply:
I don't think that is confusing, if you have a separate Pulp 'master' without a foreman-proxy(or Puppet) you don't need or want `puppet_install_distributor`.

> What happens if you move the default proxy off to another box co-located with Puppet master?

No change as, https://github.com/Katello/katello/blob/master/app/models/katello/concerns/smart_proxy_extensions.rb#L56 will still return it.

This change is purely to remove the hard requirement between foreman-proxy and Pulp master, obviously without foreman-proxy you don't get disk usage reporting ect.. but it makes the app a lot more flexible.